src/object.cr

```diff
@@ -496,7 +497,7 @@ class Object
 
         def {{method_prefix}}\{{name.id}}
           if (\%value = {{var_prefix}}\{{name.id}}).nil?
-            ::raise NilAssertionError.new("\{{name.id}} cannot be nil")
+            ::raise NilAssertionError.new("\{{@type}}" + "{{doc_prefix}}" + "\{{name.id}} cannot be nil")
```

Comment:
If anyone can find a better way of doing this, that'd be great.

Reply:
You should be able to just call the macro within the string, something like  

`"\{{@type}}{{doc_prefix}}\{{name.id}} cannot be nil"`

Reply:
I thought I could, but when I tried it, it tried to interpolate `#{{name.id}}`.

Reply:
What about `"\{{@type}}#{{{doc_prefix}}}\{{name.id}} cannot be nil"`

Reply:
Looks like you're hitting https://github.com/crystal-lang/crystal/issues/5287 you can try to use the workaround [here](https://github.com/crystal-lang/crystal/issues/5287#issuecomment-343929797) with something like:

```
"\{{@type}}\{{ "{{doc_prefix}}".id }}\{{name.id}} cannot be nil"
```

Reply:
Thank you! Was really irritating why I could escape it. I've tested it and it works perfect.